tests/sentry/event_manager/grouping/test_assign_to_group.py

```diff
@@ -0,0 +1,408 @@
+from __future__ import annotations
+
+from contextlib import contextmanager
+from time import time
+from typing import Any
+from unittest import mock
+
+import pytest
+
+from sentry.event_manager import _create_group
+from sentry.eventstore.models import Event
+from sentry.grouping.ingest import (
+    _calculate_primary_hash,
+    _calculate_secondary_hash,
+    find_existing_grouphash,
+)
+from sentry.models.grouphash import GroupHash
+from sentry.models.project import Project
+from sentry.testutils.helpers.eventprocessing import save_new_event
+from sentry.testutils.pytest.fixtures import django_db_all
+from sentry.testutils.pytest.mocking import capture_return_values
+from sentry.testutils.skips import requires_snuba
+
+pytestmark = [requires_snuba]
+
+
+LEGACY_CONFIG = "legacy:2019-03-12"
+NEWSTYLE_CONFIG = "newstyle:2023-01-11"
+
+
+@contextmanager
+def patch_grouping_helpers(return_values: dict[str, Any]):
+    wrapped_find_existing_grouphash = capture_return_values(find_existing_grouphash, return_values)
+    wrapped_calculate_primary_hash = capture_return_values(_calculate_primary_hash, return_values)
+    wrapped_calculate_secondary_hash = capture_return_values(
+        _calculate_secondary_hash, return_values
+    )
+
+    with (
+        mock.patch(
+            "sentry.event_manager.find_existing_grouphash",
+            wraps=wrapped_find_existing_grouphash,
+        ) as find_existing_grouphash_spy,
+        mock.patch(
+            "sentry.grouping.ingest._calculate_primary_hash",
+            wraps=wrapped_calculate_primary_hash,
+        ) as calculate_primary_hash_spy,
+        mock.patch(
+            "sentry.grouping.ingest._calculate_secondary_hash",
+            wraps=wrapped_calculate_secondary_hash,
+        ) as calculate_secondary_hash_spy,
+        mock.patch(
+            "sentry.event_manager._create_group",
+            wraps=_create_group,
```

Comment:
no `capture_return_values` wrapper?

Reply:
Yeah, that one I don't actually need the return values from. Really the only thing interesting/useful to know about the group in this context is its id (since that's how the grouphashes link it to hash values), and that id can be gotten off of the event. I agree it breaks the pattern, though, so it might be worth adding a comment. I can do that when I fix all the `None`s from above.